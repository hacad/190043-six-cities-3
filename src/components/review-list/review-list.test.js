import React from "react";
import renderer from "react-test-renderer";
import ReviewList from "./review-list.jsx";

it(`ReviewList correctly renders after relaunch`, () => {
  const reviews = [
    {
      "id": 1,
      "user": {
        "id": 18,
        "isPro": true,
        "name": `Sophie`,
        "avatarUrl": `https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/9.jpg`
      },
      "rating": 4,
      "comment": `The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.`,
      "date": new Date(`2019-12-03T14:11:47.471Z`)
    }
  ];

  const reviewList = renderer.create(
      <ReviewList reviews={reviews}/>
  ).toJSON();

  expect(reviewList).toMatchSnapshot();
});
