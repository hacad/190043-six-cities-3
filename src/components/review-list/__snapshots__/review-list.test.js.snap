// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewList correctly renders after relaunch when user is authorized 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews ·
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/9.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Sophie
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "4%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.
        </p>
        <time
          className="reviews__time"
          dateTime={2019-12-03T14:11:47.471Z}
        />
      </div>
    </li>
  </ul>
  &lt;div /&gt;
</section>
`;

exports[`ReviewList correctly renders after relaunch when user is not authorized 1`] = `
<section
  className="property__reviews reviews"
>
  <h2
    className="reviews__title"
  >
    Reviews ·
    <span
      className="reviews__amount"
    >
      1
    </span>
  </h2>
  <ul
    className="reviews__list"
  >
    <li
      className="reviews__item"
    >
      <div
        className="reviews__user user"
      >
        <div
          className="reviews__avatar-wrapper user__avatar-wrapper"
        >
          <img
            alt="Reviews avatar"
            className="reviews__avatar user__avatar"
            height="54"
            src="https://htmlacademy-react-2.appspot.com/six-cities/static/avatar/9.jpg"
            width="54"
          />
        </div>
        <span
          className="reviews__user-name"
        >
          Sophie
        </span>
      </div>
      <div
        className="reviews__info"
      >
        <div
          className="reviews__rating rating"
        >
          <div
            className="reviews__stars rating__stars"
          >
            <span
              style={
                Object {
                  "width": "4%",
                }
              }
            />
            <span
              className="visually-hidden"
            >
              Rating
            </span>
          </div>
        </div>
        <p
          className="reviews__text"
        >
          The deluxe room was a quite comfortable one with all the adequate facilities. The only thing that made me feel uncomfortable was the rude behavior of an impolite staff at the reception desk.
        </p>
        <time
          className="reviews__time"
          dateTime={2019-12-03T14:11:47.471Z}
        />
      </div>
    </li>
  </ul>
</section>
`;
